Task 2 (Connecting app to "Foreca" weather api)

StefanAtanasov98 opened the pull request:
Connecting the application to "Foreca" weather api includes:

- Making a simple server;
- Using AJAX to make a request for non-expiring token;
- Using AJAX to make a request for filtered list of cities;

![image](https://user-images.githubusercontent.com/103125976/194064684-88914c8f-7b86-4153-9672-e6fc91e6cc54.png)

StefanAtanasov98 committed:
Connected app to weather api

Added server, added custom hook (useAuthorize.js), fixed footer to stick to bottom

StefanAtanasov98 committed:
Made dropdown scrollable

virzhinia-petkova-redshelf commented on src/components/Homepage/Nav/DropdownMenu.js:
```diff
@@ -27,11 +30,13 @@ function DropdownMenu(props) {
   const filteredCities = searchedCity.name
     ? cities.filter(city => city.name.includes(searchedCity.name))
     : cities;
+
```
do you still need to filter cities?

virzhinia-petkova-redshelf commented on src/customHooks/useAuthorize.js:
```diff
@@ -0,0 +1,17 @@
+const useAuthorize = () => {
+  useEffect(() => {
+    (async () => {
+      if (!localStorage.getItem('token')) {
```
you have a string that you're using in different places in the app. It's very easy to do a typo and spend too much time wondering what's happening and what broke. Please move it to a constant you can use throughout the app.

virzhinia-petkova-redshelf commented on src/styles/DropdownMenu.module.css:
```diff
@@ -8,9 +8,10 @@
+  max-height: 250px;
+  overflow: auto;
 }
 
```
where did you fix your footer to stick to the bottom of the page?

StefanAtanasov98 replied:
I've set the Footer's properties "position: absolute" with "bottom: 0" and I added "position: relative" to the dropdown div. Before I made the dropdown content to display in a shorter list I tested the footer and it stuck to the bottom all the way down the cities list.

StefanAtanasov98 replied:
I also added "position: relative", "min-height: 100vh" and "margin: 0" to the html body tag

virzhinia-petkova-redshelf replied:
what do you need the position relative of the dropdown div for?

StefanAtanasov98 replied:
Well, I had problems positioning the "Cities" button itself and the dropdown content, for example when my content appears the button shrunk to half its height, so I tried a lot of things and this is what I managed. Now, when I remove "position: relative" from dropdown div, the dropdown content doesn't appear on the screen when I click the button.

virzhinia-petkova-redshelf replied:
okay

StefanAtanasov98 committed:
Cities list changes and css code refactoring

Sorted cities list by country, cleared input field for searching cities list after closing the button, fixed 'Cities' button styling

virzhinia-petkova-redshelf commented on src/components/Homepage/Nav/DropdownMenu.js:
```diff
@@ -23,15 +27,13 @@ function DropdownMenu(props) {
-    : cities;
-  const citiesList = filteredCities.map(city => (
+  
+  const citiesList = cities.sort((a, b) => a.country.localeCompare(b.country)).map(city => (
```
aren't the cities coming already sorted?

StefanAtanasov98 replied:
They aren't.. Should they be?

virzhinia-petkova-redshelf replied:
I guess I thought they were. No changes needed.

virzhinia-petkova-redshelf commented on src/customHooks/useAuthorize.js:
```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react';
+
+const URL = 'http://localhost:5020/auth';
+const token = 'token';
```
1. the convention for constants is UPPERCASE, so TOKEN
2. is this the best place to store a global variable for your app?
3. you can simply to `export const TOKEN `here

StefanAtanasov98 replied:
2. Maybe in a separate file just for global variables..?

virzhinia-petkova-redshelf replied:
sounds like a plan

StefanAtanasov98 committed:
Added .js file for global variables and changed 'token' const to uppercase
